js/authentication.md

```diff
@@ -225,6 +225,78 @@ Auth.verifyCurrentUserAttributeSubmit(attr, 'the_verification_code')
 });
 ```
 
+#### Set Session Manually
+
+You can manually set the user session by:
+```js
+import Auth, {
+    AWSCognitoProvider, 
+    GoogleProvider,
+    FacebookProvider,
+    AmazonProvider,
+    DeveloperProvider,
+    GenericProvider,
+    SessionType
+} from '@aws-amplify/auth';
+
+const { 
+    session, // the session is either a CognitoUserSession or a FederatedProviderSession
+    user, // The user is either CognitoUser or FederatedUser
+    credentials 
+} = await Auth.setSession({
+    username: 'Alice',       // Required, user name
+    attributes: {            // Optional, user attributes
+        email: 'alice@gmail.com',
+        phone_number: '123-456-7890'
+    },
+    tokens: {                 // Either id token or access token should be provided
+        idToken: 'xxxxxxx',  // Optional, the id token
+        accessToken: 'xxxxxxx', // Optional, the access token
+        refreshToken: 'xxxxxxx', // Optional, the refresh token
+        expires_at: 1540857110755, // Required, the timestamp when the token expires
+    },
+    provider: AWSCognitoProvider.NAME, // Required, the name of the token provider
+    identityId: 'xxxxxxxx', // Optional, specify the identity id from the Cognito Federated Identity Pool
+    errorHandler: (e) => {
+        // Optional, handle the error when getting credentials from Cognito Federated Identity Pool
+    },
+    credentialsDomain: 'www.example.com', // Optional, the domain used to get the credentials
+    creentialsToken: 'xxxxxxx' // Optional, the token used to get the credentials
+});
+
+// You can also get session, user, credentials from other methods:
+const session = await Auth.currentSession();
+const user = await Auth.currentAuthenticatedUser();
+const credentials = await Auth.currentCredentials();
+
+if (session.type && session.type === SessionType.Federated_Provider_Session) {
```

Comment:
could `session` be `null`?

Reply:
No, it will either return an object all throw an error.